### PR TITLE
update yed to 3.14.1

### PR DESCRIPTION
**Pull request by mjgardner**

### CODE REVIEW

File: Casks/yed.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-  sha256 '6978e180d6063e22ab0460742915b20d1f3419d284a7815041d28e318764f8ad'
+  version '3.14.1'
+  sha256 '1db07a4e108c13ed4e28ebec30d6acf4fe7b593a78ceeeed3e5a1c9d1132c512'
 
-  url "http://live.yworks.com/yed-downloads/yEd-#{version}_with-JRE8.dmg"
-  homepage 'http://www.yworks.com/en/products_yed_about.html'
-  license :unknown # todo: change license and remove this comment; ':unknown' is a machine-generated placeholder
+  url "https://www.yworks.com/products/yed/demo/yEd-#{version}_with-JRE8.dmg"
+  homepage 'http://www.yworks.com/en/products/yfiles/yed/'
+  license :gratis
```
Comment (vitorgalvao) [inline Casks/yed.rb]: Is it really `:gratis`? Asking because there's “demo” in the url.

File: Casks/yed.rb [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
+  tags :vendor => 'yWorks'
+  depends_on :macos => '>= 10.8'
```
Comment (vitorgalvao) [inline Casks/yed.rb]: This should be the last line, after `app` (with an empty line before it).

File: Casks/yed.rb (hunk 1 above)
Comment (mjgardner) [inline Casks/yed.rb]: The product page reads: "yEd is freely available and runs on all major platforms: Windows, Unix/Linux, and Mac OS X."
License is here: http://www.yworks.com/products/yed/license.html
yWorks has answered several questions about the license here: http://yed.yworks.com/support/qa/tag/license

**Comment by vitorgalvao** [comment]
Looks good, could you please [squash your commits](http://davidwalsh.name/squash-commits-git)?

**Comment by mjgardner** [comment]
I will re-do the PR as a squashed commit.

**Pull request closed without merging**